src/tensorwaves/optimizer/callbacks.py

```diff
@@ -0,0 +1,92 @@
+"""Collection of loggers that can be inserted into an optimizer as callback."""
+
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import Any, Iterable, List, Optional
+
+import tensorflow as tf
+from tqdm import tqdm
+
+
+class Callback(ABC):
+    @abstractmethod
+    def __init__(self, **kwargs: Any) -> None:
+        pass
```

Comment:
I don't like `__init__` to be part of the interface. Since that would mean the arguments (which would come from the optimizer) have to be passed at the construction of a concrete `Callback`. So I would rather change that into a normal function. Maybe like `def initialize(kwargs)`

Reply:
Oh yes, actually this is a remnant of when I still used `Type[Callback]` in the `CallbackList`, but that turned into a mess. So as it is now, this `__init__` interface can be removed without problems.

Reply:
5aa1b29